src/main/java/net/minecraftforge/event/world/BlockEvent.java

```diff
@@ -415,32 +419,57 @@ public PortalShape getPortalSize()
     }
 
     /**
-     * Fired when when this block is right clicked by a tool to change its state.
+     * Fired when this block is right-clicked by a tool to change its state.
      * For example: Used to determine if an axe can strip or a shovel can path.
-     * For hoes, see {@code net.minecraft.world.item.HoeItem#TILLABLES} and
-     * {@link net.minecraftforge.event.entity.player.UseHoeEvent}.
      *
      * This event is {@link Cancelable}. If canceled, this will prevent the tool
      * from changing the block's state.
      */
     @Cancelable
     public static class BlockToolInteractEvent extends BlockEvent
     {
-
         private final Player player;
         private final ItemStack stack;
         private final ToolAction toolAction;
+        @Nullable
+        private final UseOnContext context;
         private BlockState state;
+        private ToolModificationResult toolModificationResult = ToolModificationResult.pass(null);
 
+        /**
+         * @deprecated Use {@link #BlockToolInteractEvent(UseOnContext, BlockState, ToolAction)} instead
+         */
+        @Deprecated(forRemoval = true, since = "1.18.2")
```

Comment:
I am not sure why we should deprecate this?
Is there a thrifty reason for this?

Reply:
Because being able to post this event from a place without `UseOnContext` is wrong. All places this can be called from should have the UseOnContext. And that way, we can get rid of all the fields/helper methods in 1.19 and just return the one non-null UseOnContext.

Reply:
Okey.